### ISSUE TITLE
It does not work with latest version of ES

### ISSUE THREAD

**Issue by alkampfergit** [freemindlabsinc/FreeMindLabs.KernelMemory.Elasticsearch#35]
I've configured an ES 8.13 version in my machine, no password, no TLS.

I've configured into a minimal example of serverless memory but I got this error, also no index is created in the cluster.

```
Unhandled exception. Elastic.Transport.TransportException: Request failed to execute. Call: Status code 400 from: POST /kmdefault/_search. ServerError: Type: illegal_argument_exception Reason: "[k] must be greater than 0"
   at Elastic.Transport.DefaultHttpTransport`1.HandleTransportException(RequestData data, Exception clientException, TransportResponse response)
   at Elastic.Transport.DefaultHttpTransport`1.FinalizeResponse[TResponse](RequestData requestData, RequestPipeline pipeline, List`1 seenExceptions, TResponse response)
   at Elastic.Transport.DefaultHttpTransport`1.RequestCoreAsync[TResponse](Boolean isAsync, HttpMethod method, String path, PostData data, RequestParameters requestParameters, OpenTelemetryData openTelemetryData, CancellationToken cancellationToken)
   at Elastic.Clients.Elasticsearch.ElasticsearchClient.<>c__DisplayClass28_0`3.<<DoRequestCoreAsync>g__SendRequest|0>d.MoveNext() in /_/src/Elastic.Clients.Elasticsearch/Client/ElasticsearchClient.cs:line 158
--- End of stack trace from previous location ---
   at FreeMindLabs.KernelMemory.Elasticsearch.ElasticsearchMemory.GetSimilarListAsync(String index, String text, ICollection`1 filters, Double minRelevance, Int32 limit, Boolean withEmbeddings, CancellationToken cancellationToken)+MoveNext()
   at FreeMindLabs.KernelMemory.Elasticsearch.ElasticsearchMemory.GetSimilarListAsync(String index, String text, ICollection`1 filters, Double minRelevance, Int32 limit, Boolean withEmbeddings, CancellationToken cancellationToken)+System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult()
   at SemanticMemory.Extensions.StandardVectorSearchQueryHandler.HandleAsync(UserQuestion userQuestion, CancellationToken cancellationToken) in C:\develop\github\SemanticKernelPlayground\200_CSharpSemanticMemory\KernelMemorySamples\Extensions\StandardVectorSearchQueryHandler.cs:line 49
   at SemanticMemory.Extensions.StandardVectorSearchQueryHandler.HandleAsync(UserQuestion userQuestion, CancellationToken cancellationToken) in C:\develop\github\SemanticKernelPlayground\200_CSharpSemanticMemory\KernelMemorySamples\Extensions\StandardVectorSearchQueryHandler.cs:line 49
   at SemanticMemory.Extensions.UserQuestionPipeline.ExecuteQuery(UserQuestion userQuestion, CancellationToken cancellationToken) in C:\develop\github\SemanticKernelPlayground\200_CSharpSemanticMemory\KernelMemorySamples\Extensions\UserQuestionPipeline.cs:line 32
   at SemanticMemory.Samples.CustomPipelineBase.RunSample(String bookPdf) in C:\develop\github\SemanticKernelPlayground\200_CSharpSemanticMemory\KernelMemorySamples\Samples\CustomPipelineBase.cs:line 60
   at LogIntercepting.Program.Main(String[] args) in C:\develop\github\SemanticKernelPlayground\200_CSharpSemanticMemory\KernelMemorySamples\Program.cs:line 56
   at LogIntercepting.Program.<Main>(String[] args)

C:\develop\github\SemanticKernelPlayground\200_CSharpSemanticMemory\KernelMemorySamples\bin\Debug\net8.0\KernelMemorySamples.exe (process 70804) exited with code -532462766.
```

`

**Comment by afederici75** [comment]
Hi and sorry for the late response: I will look into this asap, but it makes me think that the limit parameter is not being set correctly. Can you verify you don't set its value to zero? https://github.com/freemindlabsinc/FreeMindLabs.KernelMemory.Elasticsearch/blob/337e58019c7d4b7754e35d15e23de04c5d181c86/src/ElasticsearchMemoryStorage/ElasticsearchMemory.cs#L227

**Comment by alkampfergit** [comment]
Hi, thanks for the answer.

Actually I rewrite a custom implementation of the memory, because I used a different approach of dynamic template, I need also to be able to search by keyword, not only do vector search. 

Actually the k element must not be set, I opened a bug in the official driver (it was fixed yesterday), because in latest version of elastic that parameter is not present anymore.

Thanks again for the answer.
